File: python/minmax.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import random


def losuj(ileliczb, maksliczb):

    liczby = []  # pusta lista

    ile = 0  # ilość unikalnych liczb

    # for i in range(ileliczb):
    while ile < ileliczb:
        liczba = random.randint(0, maksliczb)
        if liczby.count(liczba) == 0:
            liczby.append(liczba)
            ile += 1

    print(liczby)
    return liczby


def minimum(lista):
    # wyszukiwanie minimum
    min = lista[0]
    for i, el in enumerate(lista):
        if el < min:
            min = el
    return min


def maximum(lista):
    # wyszukiwanie minimum
    max = lista[0]
    for i, el in enumerate(lista):
      if el > max:
        max = el
    return max


def main(args):
    lista = losuj(10, 50)
    assert minimum([7, 5, 2, 1, 7]) == 1
    assert maximum([7, 5, 2, 1, 7]) == 7
    print("Min: ", minimum(lista))
    print("Max: ", maximum(lista))
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/minmax.cpp
/*
 * minmax.cpp
 * 
 */


#include <iostream>
#include <time.h>

using namespace std;

int losuj(int ileliczb, int maxliczb)
{
    int i, liczba;
    // int ile = 0; // ilośc unikalnych liczb
    srand(time(NULL)); // pobieranie czasu od 1970 do teraz
    for (i = 0; i < ileliczb; i++)
    {
		liczba = rand()% 40 + 10; // funkcja losująca liczbę (przedział od 10 do 50)
		cout << liczba << endl;
    }
    return liczba;
}

int minimum(int tab[])
{
	int min = tab[0];
	for( int i = 1; i < 10; i++)
	{
		if(tab[i] < min)
			min = tab[i];
	}
	return min;
}

int maximum(int tab[])
{
	int max = tab[0];
	for( int i = 1; i < 10; i++)
	{
		if(tab[i] > max)
			max = tab[i];
	}
	return max;
}

int main(int argc, char **argv)
{
	int tab[losuj(10, 50)];
	
	cout << "Min: "<< minimum(tab) << endl;
	cout << "Max: " << maximum(tab) << endl;
	
	return 0;
}
